File: cpp/Arrays and Hashing/ValidAnagram.cpp
#include <vector>
#include <iostream>
#include <set>
#include <unordered_set>
#include <unordered_map>
using namespace std;

// 242. Valid Anagram 

class Solution {
public:
   bool isAnagram(string s, string t) {
      if(s.length() != t.length()) return false;
      unordered_map<char, int> count_s;
      unordered_map<char, int> count_t;
      for(char letter : s) count_s[letter] += 1;
      for(char letter : t) count_t[letter] += 1;
      return (count_s==count_t);
   }  
};

File: Python/Arrays/valid_anagram.py
#242. Valid Anagram
from typing import List
from collections import defaultdict

class Solution:
   def isAnagram(self, s: str, t: str) -> bool:
      counter_s = defaultdict(int)
      counter_t = defaultdict(int)

      for letter in s:
         counter_s[letter] += 1
         
      for letter in t:
         counter_t[letter] += 1

      return counter_t == counter_s